flytekit/engines/unit/engine.py

```diff
@@ -251,15 +251,18 @@ def _transform_for_user_output(self, outputs):
         """
         futures = outputs.get(_sdk_constants.FUTURES_FILE_NAME)
         if futures:
+            queries = []
             task_ids_to_defs = {
                 t.id.name: _qubole_models.QuboleHiveJob.from_flyte_idl(
                     _ParseDict(t.custom, _qubole_pb2.QuboleHiveJob())
                 )
                 for t in futures.tasks
             }
-            return [
-                q.query
-                for q in task_ids_to_defs[futures.nodes[0].task_node.reference_id.name].query_collection.queries
-            ]
+            for node in futures.nodes:
+                queries.extend(
+                    q.query
+                    for q in task_ids_to_defs[node.task_node.reference_id.name].query_collection.queries
```

Comment:
can we just use the `.query.query` field?  it should be there.  We're still creating one element `QueryCollection`'s just for backwards compatibility.